source/common/http/http1/codec_impl.cc

```diff
@@ -351,14 +360,12 @@ void ResponseEncoderImpl::encodeHeaders(const ResponseHeaderMap& headers, bool e
   connection_.addCharToBuffer('\r');
   connection_.addCharToBuffer('\n');
 
-  if (numeric_status == 204 || numeric_status < 200) {
-    // Per https://tools.ietf.org/html/rfc7230#section-3.3.2
-    setIsContentLengthAllowed(false);
-  } else {
-    // Make sure that if we encodeHeaders(100) then encodeHeaders(200) that we
-    // set is_content_length_allowed_ back to true.
-    setIsContentLengthAllowed(true);
-  }
+  // Enabling handling of https://tools.ietf.org/html/rfc7230#section-3.3.1 and
+  // https://tools.ietf.org/html/rfc7230#section-3.3.2. Also resets these flags
+  // if a 100 Continue is followed by another status.
+  setIs1xx(numeric_status < 200);
```

Comment:
I vaguely recall asking @alyssawilk this when the above code was written, but can't we just pass some state directly into `encodeHeadersBase` such as `bool never_chunk_encode` and avoid the indirection with member variables? It seems simpler to understand but I don't feel strongly about it. The base function already has request/response specific logic also so it seems not even terrible to just pass the is_1xx_response and is_204_response directly also, but a generic bool seems better.